app/uk/gov/hmrc/cgtpropertydisposalsfrontend/repos/Repo.scala

```diff
@@ -36,27 +34,7 @@ trait Repo {
     id: String
   )(implicit ec: ExecutionContext): Future[Either[Error, Option[A]]] =
     preservingMdc {
-      cacheRepository
-        .findById(id)
-        .map { maybeCache =>
-          val response: OptionT[Either[Error, *], A] = for {
-            cache  <- OptionT.fromOption[Either[Error, *]](maybeCache)
-            data   <- OptionT.fromOption[Either[Error, *]](Some(cache.data))
-            result <- OptionT.liftF[Either[Error, *], A](
-                        (data \ sessionKey)
-                          .validate[A]
-                          .asEither
-                          .leftMap(e =>
-                            Error(
-                              s"Could not parse session data from mongo: ${e.mkString("; ")}"
-                            )
-                          )
-                      )
-          } yield result
-
-          response.value
-        }
-        .recover { case e => Left(Error(e)) }
+      EitherT.liftF(cacheRepository.get(id)(DataKey(sessionKey))).value
```

Comment:
💡 You could be cheeky and avoid EitherT altogether:
```scala
      cacheRepository.get(id)(DataKey(sessionKey)).map(Right(_))
```

Reply:
Yeah, I was thinking that based on what is below but completely forgot to try it. Will try that now